### PR TITLE
Test native `segment_csr` behavior, add documentation for integralTransform to website

### PR DESCRIPTION
**Pull request by dhpitt**
Adds tests for `segment_csr`, cleans up docstrings, updates the trees for autogenerated Sphinx docs website, changes the forward call signature for a `GINO` to take `x` instead of `f` for maximum compatibility.

### CODE REVIEW

File: doc/source/modules/api.rst [hunk 1]
```diff
@@ -126,6 +126,34 @@ Neural operator Layers
 
     SphericalConv
 
+**Kernel integration**
```
Comment (JeanKossaifi) [inline doc/source/modules/api.rst:129]: Perhaps we add some text around to example the high level goal?

File: doc/source/modules/api.rst [hunk 2]
```diff
@@ -126,6 +126,34 @@ Neural operator Layers
...
+    :template: function.rst
+
+    simple_neighbor_search
+
+.. autosummary::
+    :toctree: generated
+    :template: function.rst
+
+    segment_csr
+
```
Comment (JeanKossaifi) [inline doc/source/modules/api.rst:157]: Should we also add GINO to the API?

**Review by JeanKossaifi — COMMENTED** [review]
Looks good! I think we can just add some text in the API and add GINO too.

File: doc/source/modules/api.rst (hunk 2 above)
Comment (dhpitt) [reply doc/source/modules/api.rst:157]: It should be autogenerated in `models`

**Comment by JeanKossaifi** [comment]
Looks like there's a typo that's breaking the tests.

**Comment by JeanKossaifi** [comment]
Looks good, thanks David!

**Pull request merged**